Add `rename_enum` method for `pg_enum` extension

AlexWayfer opened the pull request.

jeremyevans commented:
Thanks for the patch. These changes look good, but can you also add a spec to `specs/adapters/postgres.rb`?

celsworth commented:
Is there any chance this could be made reversible too? Seems like all the information (old name, new name) will be present.

AlexWayfer commented:
> These changes look good, but can you also add a spec to specs/adapters/postgres.rb?

Sure, I didn't think this is necessary for new method. Added.

> Is there any chance this could be made reversible too? Seems like all the information (old name, new name) will be present.

Good point! I added this, thank you.
